hyperspy/tests/signals/test_1D_tools.py

```diff
@@ -218,11 +218,26 @@ def test_single_spectrum(self):
         np.testing.assert_array_equal(s.data, np.arange(20))
         assert m.data_changed.called
 
+    def test_single_spectrum_nonuniform(self):
```

Comment:
Since the expected results are exactly the same, would it make sense to use `pytest.mark.parametrize('uniform', [True, False])` and add a `if` block in the test to convert to a non-uniform axis? Same for the other test?